src/java.base/share/classes/java/util/Formatter.java

```diff
@@ -4490,7 +4490,14 @@ private Appendable print(Formatter fmt, StringBuilder sb, TemporalAccessor t, ch
                 }
                 case DateTime.ISO_STANDARD_DATE: { // 'F' (%Y-%m-%d)
                     char sep = '-';
-                    print(fmt, sb, t, DateTime.YEAR_4, l).append(sep);
+                    int year = t.get(ChronoField.YEAR);
+                    if (year < 0) {
+                        sb.append('-');
```

Comment:
Some locales use a different code point to express negative numbers from `hyphen-minus`, such as `minus sign` (U+2212). I'd suggest getting it from `DecimalFormatSymbols.getMinusSign()`